neo4j/src/main/java/eu/drus/jpa/unit/neo4j/Neo4JDbDecorator.java

```diff
@@ -0,0 +1,53 @@
+package eu.drus.jpa.unit.neo4j;
+
+import java.sql.Connection;
+
+import javax.sql.DataSource;
+
+import eu.drus.jpa.unit.neo4j.ext.ConfigurationRegistry;
+import eu.drus.jpa.unit.spi.ExecutionContext;
+import eu.drus.jpa.unit.spi.TestInvocation;
+import eu.drus.jpa.unit.spi.TestMethodDecorator;
+
+public class Neo4JDbDecorator implements TestMethodDecorator {
+
+    private ConfigurationRegistry configurationRegistry = new ConfigurationRegistry();
+
+    @Override
+    public int getPriority() {
+        return 5;
+    }
+
+    @Override
+    public boolean isConfigurationSupported(final ExecutionContext ctx) {
+        return configurationRegistry.hasConfiguration(ctx.getDescriptor());
+    }
+
+    @Override
+    public void beforeTest(final TestInvocation invocation) throws Exception {
+        final ExecutionContext context = invocation.getContext();
+
+        final Neo4JDbFeatureExecutor dbFeatureExecutor = new Neo4JDbFeatureExecutor(invocation.getFeatureResolver(),
+                context.getDescriptor().getClasses());
+        final DataSource ds = (DataSource) context.getData(Constants.KEY_DATA_SOURCE);
+        final Connection connection = ds.getConnection();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "Connection" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)

Reply:
The life cycle management of a connection is spread across before- and afterTest hook methods. The proposal cannot be applied here.